test/plots/rectanglePlotTests.ts

```diff
@@ -183,289 +198,233 @@ describe("Plots", () => {
+          const rect = d3.select(this);
+          const rectCenterX = TestMethods.numAttr(rect, "x") + TestMethods.numAttr(rect, "width") / 2;
+          const rectCenterY = TestMethods.numAttr(rect, "y") + TestMethods.numAttr(rect, "height") / 2;
+          assert.closeTo(xScale.scale(d.x), rectCenterX,
+            window.Pixel_CloseTo_Requirement, `center of rect ${i} in x is scaled version of input data`);
+          assert.closeTo(yScale.scale(d.y), rectCenterY,
+            window.Pixel_CloseTo_Requirement, `center of rect ${i} in y is scaled version of input data`);
+        });
         svg.remove();
       });
 
       it("renders correctly when there isn't data for every spot", () => {
-        let dataset = new Plottable.Dataset();
-        let plot = new Plottable.Plots.Rectangle();
-        plot.addDataset(dataset);
-        plot.attr("fill", (d) => d.magnitude, colorScale);
-        plot.x((d: any) => d.x, xScale);
-        plot.y((d: any) => d.y, yScale);
-        plot.renderTo(svg);
-        let data = [
-          {x: "A", y: "W", magnitude: 0},
-          {x: "B", y: "X", magnitude: 8},
-          {x: "C", y: "Y", magnitude: 16},
-          {x: "D", y: "Z", magnitude: 24}
+        const data = [
+          {x: "A", y: "W"},
+          {x: "B", y: "X"},
+          {x: "C", y: "Y"},
+          {x: "D", y: "Z"}
         ];
-        dataset.data(data);
-        let cells = plot.content().selectAll("rect")[0];
-        assert.strictEqual(cells.length, data.length);
-        let cellHeight = 50;
-        let cellWidth = 100;
-        for (let i = 0; i < cells.length; i++) {
-          let cell = d3.select(cells[i]);
-          assert.strictEqual(cell.attr("x"), String(i * cellWidth), "Cell x coord is correct");
-          assert.strictEqual(cell.attr("y"), String(i * cellHeight), "Cell y coord is correct");
-          assert.strictEqual(cell.attr("width"), String(cellWidth), "Cell width is correct");
-          assert.strictEqual(cell.attr("height"), String(cellHeight), "Cell height is correct");
-        }
+        plot.addDataset(new Plottable.Dataset(data));
+        plot.renderTo(svg);
+
+        const rects = plot.content().selectAll("rect");
+        assert.strictEqual(rects.size(), data.length, "rect for each datum");
+
+        rects.each(function(d, i) {
+          const rect = d3.select(this);
+          const rectCenterX = TestMethods.numAttr(rect, "x") + TestMethods.numAttr(rect, "width") / 2;
+          const rectCenterY = TestMethods.numAttr(rect, "y") + TestMethods.numAttr(rect, "height") / 2;
+          assert.closeTo(xScale.scale(d.x), rectCenterX,
+            window.Pixel_CloseTo_Requirement, `center of rect ${i} in x is scaled version of input data`);
+          assert.closeTo(yScale.scale(d.y), rectCenterY,
+            window.Pixel_CloseTo_Requirement, `center of rect ${i} in y is scaled version of input data`);
+        });
         svg.remove();
       });
 
       it("can invert y axis correctly", () => {
```

Comment:
It looks like this test checks that the code works correctly if the domain is reversed? Can we rename it so that's clearer?